### PR TITLE
test: add a unit test when podman is not present

### LINKED ISSUE #782: Add a unit test in `DevOnPodmanComponentActionTest`
Maybe you could add a unit test in `DevOnPodmanComponentActionTest` testing enablement when podman is present/not present?

_Originally posted by @adietish in https://github.com/redhat-developer/intellij-openshift-connector/pull/773#discussion_r1562585674_

### PR DESCRIPTION
**Pull request by sbouchet**
## Was the change discussed in an issue?
fixes #782

### CODE REVIEW

**Review by adietish — APPROVED** [review]
found a few nitpicks. Other than that things look good to me.

File: src/it/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCliComponentTest.java [hunk 1]
```diff
@@ -45,26 +45,31 @@ public OdoCliComponentTest(ComponentFeature feature, String label) {
   public static Iterable<Object[]> data() {
     return Arrays.asList(new Object[][]{
       {ComponentFeature.DEV, ComponentFeature.DEV.getLabel()},
-      {ComponentFeature.DEV_ON_PODMAN, ComponentFeature.DEV_ON_PODMAN.getLabel()}
+      {ComponentFeature.DEV_ON_PODMAN, ComponentFeature.DEV_ON_PODMAN.getLabel()},
```
[the remark this answers is not in the document]
Comment (sbouchet) [reply src/it/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCliComponentTest.java]: indeed a  typo. good catch !

File: src/it/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCliComponentTest.java [hunk 2]
```diff
@@ -82,16 +87,20 @@ public void checkCreateComponent() throws IOException, ExecutionException, Inter
     descriptors = odo.discover(projectPath);
     assertNotNull(descriptors);
     assertEquals(1, descriptors.size());
+    //cleanup
+    odo.deleteComponent(project, projectPath, component, ComponentKind.DEVFILE);
```
[the remark this answers is not in the document]
Comment (sbouchet) [reply src/it/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCliComponentTest.java]: refactored the cleanup() to stop and delete component if still exists

File: src/it/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCliComponentTest.java [hunk 3]
```diff
@@ -111,44 +120,72 @@ public void checkCreateComponentAndLinkService() throws IOException, ExecutionEx
...
     int debugPort;
     try (ServerSocket serverSocket = new ServerSocket(0)) {
       debugPort = serverSocket.getLocalPort();
     }
     ExecHelper.submit(() -> {
       try {
         odo.debug(projectPath, debugPort);
-        assertTrue(odo.isStarted(component, feature));
       } catch (IOException e) {
         fail("Should not raise Exception");
```
[the remark this answers is not in the document]
Comment (sbouchet) [reply src/it/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCliComponentTest.java]: fixed.

**Comment by adietish** [comment]
@sbouchet: I dont understand what's wrong with the `Conventional Commits PR Check / Cluster Integration UI Tests Report (pull_request_target)`

**Comment by sbouchet** [comment]
> @sbouchet: I dont understand what's wrong with the `Conventional Commits PR Check / Cluster Integration UI Tests Report (pull_request_target)`

it means the Cluster Integration UI Tests runs failed.

**Pull request merged**